src/design-system/organisms/SwapForm/SwapForm.tsx

```diff
@@ -0,0 +1,354 @@
+  const PositionDescription = () => (
+    <>
+      {t('expectedAdjustedPosition')}
+      <IconTooltip
+        description="Swap rate is discounted by 0.3% fee. This fee goes to the liquidity providers."
+        placement="bottom-start"
+        maxWidth={theme.spacing(31)}
+      />
+    </>
+  );
+  React.useEffect(() => {
+    if (poolTokens) {
+      const yesPool = getTokenQuantityById(poolTokens, yesTokenId);
+      const noPool = getTokenQuantityById(poolTokens, noTokenId);
+      setPools({
+        yesPool,
+        noPool,
+      });
+    }
+    if (userTokens) {
+      const yesToken = getTokenQuantityById(userTokens, yesTokenId);
+      const noToken = getTokenQuantityById(userTokens, noTokenId);
+      setUserAmounts({
+        noToken,
+        yesToken,
+      });
+    }
+  }, [poolTokens, userTokens, yesTokenId, noTokenId]);
+
+  const validationSchema = React.useMemo(() => {
+    if (swapTokenType === TokenType.yes) {
+      if (connected && userBalance) {
+        return Yup.object({
+          yesToken: Yup.number()
+            .min(0.000001, `${t('minSwap')} 0.000001`)
+            .max(userBalance, t('insufficientBalance'))
+            .required(t('required')),
+        });
+      }
+    } else if (connected && userBalance) {
+      return Yup.object({
+        noToken: Yup.number()
+          .min(0.000001, `${t('minSwap')} 0.000001`)
+          .max(userBalance, t('insufficientBalance'))
+          .required(t('required')),
+      });
+    }
+    return Yup.object({
+      yesToken: Yup.number().max(0, t('insufficientBalance')),
+      noToken: Yup.number().max(0, t('insufficientBalance')),
+    });
+  }, [userAmounts, connected, userBalance]);
+
+  const handleChange = React.useCallback(
+    (e: any) => {
+      if (Number.parseFloat(e.target.value) > 0) {
+        const [aPool, bPool] =
+          TokenType.yes === swapTokenType
+            ? [pools.yesPool, pools.noPool]
+            : [pools.noPool, pools.yesPool];
+        const quantity = tokenMultiplyUp(e.target.value);
+        const { swapOutput, exchangeRate } = swapTokenCalculations(
+          quantity,
+          aPool,
+          bPool,
+          slippage,
+        );
+
+        const newExpectedSwap: PositionItem[] = [
+          {
+            label: t('expectedRate'),
+            value: `1 ${t(swapToken)} = ${roundToTwo(tokenDivideDown(exchangeRate))} ${t(
+              otherToken,
+            )}`,
+          },
+          {
+            label: t('output'),
+            value: `${roundToTwo(tokenDivideDown(swapOutput))} ${t(otherToken)}`,
+          },
+        ];
+        setExpectedSwap(newExpectedSwap);
+
+        if (connected) {
+          const [newYes, newNo, yesOpt, noOpt] =
+            TokenType.yes === swapTokenType
+              ? [quantity, swapOutput, '-', '+']
+              : [swapOutput, quantity, '+', '-'];
+
+          let newTotalValue;
```

Comment:
i think i prefer this as a ternary expression using a const for setting either value depending on the tokenType, but I'm not sure if its 'better'

Reply:
example: 
```suggestion
          const newTotalValue= TokenType.yes? tokenPrice.yes * (userAmounts.yesToken - newYes) +
              tokenPrice.no * (userAmounts.noToken + newNo):tokenPrice.yes * (userAmounts.yesToken + newYes) +
              tokenPrice.no * (userAmounts.noToken - newNo);
```

Reply:
or better as a function with tokenType as a parameter as you suggested!